File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/model/memory/MemoryExtractionContext.java
package com.study.webflux.rag.domain.model.memory;

import com.study.webflux.rag.domain.model.conversation.ConversationTurn;
import java.util.List;

public record MemoryExtractionContext(
	List<ConversationTurn> recentConversations,
	List<Memory> existingMemories
) {
	public MemoryExtractionContext {
		if (recentConversations == null) {
			recentConversations = List.of();
		}
		if (existingMemories == null) {
			existingMemories = List.of();
		}
	}

	public static MemoryExtractionContext of(List<ConversationTurn> conversations,
		List<Memory> memories) {
		return new MemoryExtractionContext(conversations, memories);
	}
}

File: app-miyou/src/main/kotlin/com/miyou/app/domain/memory/model/MemoryExtractionContext.kt
package com.miyou.app.domain.memory.model

data class ConversationSnippet(
    val query: String,
    val response: String?,
)

data class MemoryExtractionContext(
    val sessionId: String,
    val recentConversations: List<ConversationSnippet> = emptyList(),
    val existingMemories: List<Memory> = emptyList(),
) {
    companion object {
        @JvmStatic
        fun of(
            sessionId: String,
            conversations: List<ConversationSnippet>,
            memories: List<Memory>,
        ): MemoryExtractionContext = MemoryExtractionContext(sessionId, conversations, memories)
    }
}
